realm/realm-library/src/main/cpp/CMakeLists.txt

```diff
@@ -36,10 +36,24 @@ set_target_properties(lib_realm_core PROPERTIES IMPORTED_LOCATION
 
 # build application's shared lib
 include_directories(${distribution_DIR}/realm_core/include ${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/jni_include)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
+
+# Set compile flags
+if (${ANDROID_ABI} STREQUAL "armeabi")
+    set(ABI_CXX_FLAGS "-mthumb")
+elseif (${ANDROID_ABI} STREQUAL "armeabi-v7a")
+    set(ABI_CXX_FLAGS "-mthumb -march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16")
+endif()
+set(WARNING_CXX_FLAGS "-Wall -Wextra -pedantic -Wno-long-long -Wno-variadic-macros ")
+set(REALM_COMMON_CXX_FLAGS " -DREALM_HAVE_CONFIG -DPIC -pthread -fvisibility=hidden -std=gnu++11")
```

Comment:
It is part of default android cmake flags. But maybe we should just overwrite them all.